### PR TITLE
Create chess pieces.

### PR DESCRIPTION
**Pull request by Felizardo**
As discussed in #48. This PR only creates the chess pieces, I will send another one after this with the board.

### CODE REVIEW

**Comment by nicolodavis** [comment]
Great work, @Felizardo! Thanks for getting everything tested as well.

Just want some changes before we can get this in:

#### 1.  The vision for tokens in boardgame.io is to have exactly one `Token` component that can accept other sub-components for rendering purposes:

The sub-component will actually be optional (the token will render as a blank square by default). This will be great for prototyping.

```
<Token render={Knight} />
```

This token component is what we will expand (eventually) to add drag-n-drop, hover animations etc. The `Knight` implementation should be fairly dumb (just rendering without any behavior attached to it).

In your case, I guess just rename `ChessPiece` to `Token`, remove anything chess-specific and move it further up the directory hierarchy?

#### 2. The user should not interact with anything SVG.

This is an implementation detail that the user should not have to worry about. I want us to have our own co-ordinate system so we can implement generic grids and hex grids. The underlying implementation can be SVG or something else, but should be hidden from the user.

```
<Grid rows={8} cols={8}>
   <Token x={0} y={0} render={Knight}/>
</Grid>
```

should create a grid and place the token on the top left square. This should also work seamlessly when using a `HexGrid`

```
<HexGrid ...>
   <Token x={0} y={0} z={0} ...>
</HexGrid>
```

Implementing `HexGrid` is outside the scope of this PR, but perhaps you can create a tiny `Grid` implementation so that you can get rid of SVG parameters from the props everywhere?

File: src/client/lib/chess/piece.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      x: this.props.x,
+      y: this.props.y
+    });
+  }
+
+  /**
+  * If there is a change in x and y, saves old X, Y, and current time. Starts
+  * animation.
+  */
+  componentWillReceiveProps (nextProps) {
```
Comment (nicolodavis) [inline src/client/lib/chess/piece.js]: No space before the paren just to be consistent with the code style elsewhere. Can you also add https://eslint.org/docs/rules/space-before-function-paren to the .eslintrc file?

**Comment by Felizardo** [comment]
Oh, I see, the Token component would handle all the behavior then, and the
subcomponents (King and etc) the presentation, right? That makes total
sense.

WRT grid, then I could encapsulate the <svg> element in it, for now? We
probably need to specify some attributes for rendering (width, height) of
the element too, right?

**Comment by Felizardo** [comment]
Is the grid supposed to behave like the CSS grid?
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout

**Comment by nicolodavis** [comment]
The Grid would be much simpler than a CSS grid. Just a matrix with rows and columns of fixed length, and an integer co-ordinate system to identify locations. Tokens map to a single location on the Grid (they cannot span multiple ones).

The Grid can encapsulate SVG elements of course. Just make sure that the external interface does not refer to anything that is specific to SVG (width and height are ok).

**Comment by jchristman** [comment]
One little comment I have for Token is that I think it may be more natural for React developers for it to follow a more typical HoC usage syntax. That is, using it may be more along the lines of:

```
import { Token } from ‘boardgame.io/something’;

const options = {
  // possibly not needed
};

const Knight = (props) => (
  // any special rendering parts here. X and y and other get passed as props
);

export default Token(options)(Knight);
```

Sorry I’m on my phone so may not be entirely clear, but I mostly don’t like passing components as props...

**Comment by nicolodavis** [comment]
The HoC approach sounds fine too. However, Token should be usable by itself. I fear that making a HoC will give the impression that it is always meant to be used as a wrapper around another component.

**Comment by Felizardo** [comment]
Ok, I was going to suggest child-component too :)

**Comment by Felizardo** [comment]
I am closing this pull request, I am going to split this code into two, one for the Token, Grid and one later for Chess example.

**Pull request closed without merging**